Student/Student/Dashboard/Views/DashboardCourseCell.swift

```diff
@@ -84,16 +85,45 @@ class DashboardCourseCell: UICollectionViewCell {
         optionsCallback = nil
     }
 
-    func configure(with model: Course) {
+    func configure(with model: Course, hideDashcardColorOverlays: Bool) {
         let color = model.color.ensureContrast(against: .named(.white))
         course = model
         titleLabel?.text = model.name
         titleLabel?.textColor = color
         abbrevationLabel?.text = model.courseCode
-        topView?.backgroundColor = color
+
+        if (model.showColorOverlay(hideOverlaySetting: hideDashcardColorOverlays)) {
+            configureWithOverlay(color: color)
+        } else {
+            configureWithoutOverlay(color: color)
+        }
+
         imageView?.load(url: model.imageDownloadURL)
     }
 
+    func configureWithOverlay(color: UIColor) {
+        topView?.backgroundColor = color
+        imageView?.alpha = 0.4
+        optionsButton?.backgroundColor = .clear
+        optionsButton?.layer.cornerRadius = 0
+        optionsButtonCircle?.removeFromSuperlayer()
+    }
+
+    func configureWithoutOverlay(color: UIColor) {
+        topView?.backgroundColor = .white
+        imageView?.alpha = 1
+
+        guard let optionsButton = optionsButton else {
+            return
+        }
+        let circleLayer = CALayer()
+        circleLayer.frame = CGRect(x: 4, y: 4, width: optionsButton.bounds.width - 8, height: optionsButton.bounds.height - 8)
+        circleLayer.backgroundColor = color.cgColor
+        circleLayer.cornerRadius = circleLayer.frame.width / 2
+        optionsButton.layer.insertSublayer(circleLayer, below: optionsButton.imageView?.layer)
+        optionsButtonCircle = circleLayer
```

Comment:
do we need to create a new layer, or can the button just have a background color & it's normal layer get rounded?

Reply:
I initially went that route but the problem was that the button's size was larger than what we wanted the visual size of the circle to be.

Reply:
it likely doesn't need to be any bigger than the circle is. We probably were just going for a nice hit area, which may not be necessary.

Reply:
I'll chat with David to see what he thinks. The circle is only 26x26 which is a bit small, and the button is 34x34 which still seems small but I'll see what he thinks.

Reply:
My intent was to maintain the larger hit target while visually having it match the designs.